### ISSUE TITLE
Day 4: Modularize Training Loop into `train_utils.training_loop`

### ISSUE THREAD

**Issue by hmehryar** [hmehryar/Hm.JetscapeMl#354]
**Description:**  
Move the main training logic from `train.ipynb` into a reusable function within the `train_utils` namespace to promote clarity and reuse across experiments.

**Checklist:**
- [x] Create `train_utils/training_loop.py` inside `experiment/exp_25072401_hybrid_mamaba_vit_statck/`
- [x] Define `run_training_loop(model, train_loader, val_loader, test_loader, cfg)` that:
  - Trains the model for `cfg.epochs`
  - Tracks training/validation loss and metrics
  - Applies early stopping and LR scheduler
  - Saves `summary.json` and plots to `cfg.output_dir`
  - Returns `(model, summary, test_metrics)`
- [x] Replace in-notebook training loop with:
  ```python
  from train_utils.training_loop import run_training_loop
  model, summary, test_metrics = run_training_loop(model, train_loader, val_loader, test_loader, cfg)
